### ISSUE TITLE
VoiceOver (macOS) wcag issues for ImageUpload component

### ISSUE THREAD

**Issue by lassopicasso** [Altinn/app-frontend-react#3815]
### Description of the bug

KS Digital discovered several WCAG-related issues when testing the new `ImageUpload` component using VoiceOver on Mac.

**(1) Focus is lost after selecting an image**
- After choosing a file, focus falls back to the browser window, not the component.
- VoiceOver navigation (`VO+Right`) stops working until manually pressing `Tab`.
- Expected behavior: focus should return to the Canvas where the image is displayed in the image upload component (similar to `FileUpload`).

**(2) Focus jumps to the top of the page when saving an image**
- When saving an uploaded image, focus moves to the top of the page rather than staying on the component.
- Expected behavior: After saving, focus should be on the delete button.

**(3) "Change Image" button has inconsistent focus**
- VoiceOver selects different parts of the button depending on direction of navigation:
   - Left to Right: selects icon
   - Right to left: selection text
- Likely caused by our current implementation where a hidden `<input type="file">` is linked to a `<label>` that is inside a `<Button asChild>`.
**This structure was originally chosen to support keyboard activation via space/enter, but it seems to cause unpredictable focus behavior in macOS screen reader.
```
<Input
  id={inputId}
  ref={fileInputRef}
  type='file'
  accept='image/*'
  onChange={handleImageChange}
  hidden
/>

<Button
  asChild
  data-size='sm'
  variant='secondary'
  data-color='accent'
  tabIndex={0}
  onKeyDown={(e) => handleFileSelectKeyDown(e)}
>
  <Label htmlFor={inputId}>
    <UploadIcon />
    <Lang id='image_upload_component.button_change' />
  </Label>
</Button>
```

### Steps To Reproduce

KS Digital demonstration (VoiceOver on Mac):

https://github.com/user-attachments/assets/00750720-b387-4968-9cec-d23c72d34415

Correct behavior in FileUpload:

https://github.com/user-attachments/assets/f7bc2884-182a-4ea7-816a-927820789dcd

### Additional Information

There is a Slack thread with additional context (in locked channel).
Let me know if you want access or have any other questions:
https://digdir-samarbeid.slack.com/archives/C090VFARF1V/p1761046046934189
